### PR TITLE
ruby: Add size to RepeatField

### PR DESCRIPTION
**Pull request by skippy**
initial steps to make RepeatedField act more like a ruby Array

Let me know if this approach of aliasing and adding syntactic sugar in the ruby lib, rather than duplicating the logic in C and Java, is acceptable.

### CODE REVIEW

File: ruby/lib/google/protobuf/repeated_field.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#add syntatic sugar on top of the core library
```
Comment (cfallin) [inline ruby/lib/google/protobuf/repeated_field.rb]: Tiny nitpick: space between the `#` start of comment and the comment text?

**Comment by skippy** [comment]
related to #320

**Comment by cfallin** [comment]
Looks good; could you squash to a single commit? Thanks!

**Comment by skippy** [comment]
np!

**Comment by cfallin** [comment]
Thanks, merging.

**Pull request merged**